xbmc/utils/EGLUtils.cpp

```diff
@@ -655,3 +652,51 @@ bool CEGLContextUtils::HasContext()
 {
   return eglGetCurrentContext() != EGL_NO_CONTEXT;
 }
+
+void CEGLContextUtils::SetDamagedRegions(const CDirtyRegionList& dirtyRegions)
+{
+  if (!m_partialUpdateSupport)
+    return;
+
+  using Rect = std::array<EGLint, 4>;
+  if (dirtyRegions.empty())
+  {
+    // add a single (empty) entry, otherwise the whole frame gets rendered
+    static Rect zeroRect{};
+    m_eglSetDamageRegionKHR(m_eglDisplay, m_eglSurface, zeroRect.data(), 1);
+  }
+  else
+  {
+    EGLint height = eglQuerySurface(m_eglDisplay, m_eglSurface, EGL_HEIGHT, &height);
+    std::vector<Rect> rects;
+    rects.reserve(dirtyRegions.size());
+    for (const auto& region : dirtyRegions)
+    {
+      rects.push_back({static_cast<EGLint>(region.x1), static_cast<EGLint>(height - region.y2),
```

Comment:
emplace_back?

Reply:
That's not possible, is it? At least not without having a local `Rect` in the loop to emplace. Or so my understanding.

Does this make a difference?

Reply:
> That's not possible, is it? At least not without having a local Rect in the loop to emplace. Or so my understanding.

You are correct, you can't `emplace_back` a `std::array` even in C++20 where it supports `emplace_back` for aggregates.